project/project.go

```diff
@@ -28,24 +28,26 @@ type Project struct {
 	ReloadCallback func() error
 	ParseOptions   *config.ParseOptions
 
-	runtime       RuntimeProject
-	configVersion string
-	context       *Context
-	reload        []string
-	upCount       int
-	listeners     []chan<- events.Event
-	hasListeners  bool
+	runtime         RuntimeProject
+	networksFactory NetworksFactory
+	configVersion   string
+	context         *Context
+	reload          []string
+	upCount         int
+	listeners       []chan<- events.Event
+	hasListeners    bool
 }
 
 // NewProject creates a new project with the specified context.
-func NewProject(context *Context, runtime RuntimeProject, parseOptions *config.ParseOptions) *Project {
+func NewProject(context *Context, runtime RuntimeProject, networksFactory NetworksFactory, parseOptions *config.ParseOptions) *Project {
```

Comment:
It seems a little odd that `serviceFactory` is in `context` but `networksFactory` isn't. Not sure which of the two places is better, but we should probably be consistent.